packages/cli/src/index.ts

```diff
@@ -359,20 +373,47 @@ program
     'The maximum value (in gwei) for the miner tip when submitting the registry transaction'
   )
   .option('-q --quiet', 'Only output final JSON object at the end, no human readable output')
+  .option('--include-provisioned', 'Includes provisioned packages when publishing to the registry')
   .action(async function (packageRef, options) {
     const { publish } = await import('./commands/publish');
 
-    const cliSettings = resolveCliSettings(options);
-    const p = await resolveRegistryProvider(cliSettings);
+    if (!options.chainId) {
+      const chainIdPrompt = await prompts({
+        type: 'number',
+        name: 'value',
+        message: 'Please provide a Chain ID',
+        initial: 13370,
+      });
 
-    const overrides: ethers.Overrides = {};
+      if (!chainIdPrompt.value) {
+        console.log('Chain ID not provided.');
+        process.exit(1);
+      }
 
-    if (!options.chainId) {
-      throw new Error(
-        'Please provide a chainId using the format: --chain-id <number>. For example, 13370 is the chainId for a local build.'
-      );
+      options.chainId = chainIdPrompt.value;
+    }
+
+    if (!options.privateKey && !process.env.PRIVATE_KEY) {
+      const keyPrompt = await prompts({
+        type: 'text',
+        name: 'value',
+        message: 'Please provide a Private Key',
+        style: 'password',
+      });
+
+      if (!keyPrompt.value) {
```

Comment:
can we do better verification here? for example, check to make suer the value is 66 characters long and is a hex string starting with 0x, or 64 characters long and a hex string *not* starting with 0x. ethers probably has a function to help with this.

Reply:
Absolutely, good catch. Thanks for bringing this up.